db/migrate/20160513160206_initial_migration.rb

```diff
@@ -0,0 +1,34 @@
+class InitialMigration < ActiveRecord::Migration
+  def change
+    create_table :links do |t|
+      t.string :uri
+      t.integer :upvote_count
+      t.integer :downvote_count
+
+      t.timestamps null: false
+    end
+
+    create_table :tags do |t|
+      t.string :name
```

Comment:
Ummmm probably not. Real question is how you think a table is going to be searched. 

The name of the tags table is likely going to be used in nearly every query involving a tag, so it makes sense to add the index. 

The URI is probably only going to be used after a search from some other datapoint, via a relation, so probably don't need to index it.